cmake/compiler/iar/compiler_flags.cmake

```diff
@@ -0,0 +1,167 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Compiler options for the IAR C/C++ Compiler for Arm
+
+
+#####################################################
+# This section covers flags related to optimization #
+#####################################################
+set_compiler_property(PROPERTY no_optimization -On)
+
+set_compiler_property(PROPERTY optimization_debug -Ol)
+
+set_compiler_property(PROPERTY optimization_speed -Ohs)
+
+set_compiler_property(PROPERTY optimization_size -Ohz)
+
+#######################################################
+# This section covers flags related to warning levels #
+#######################################################
+
+# Property for standard warning base in Zephyr, this will always be set when
+# compiling.
+set_compiler_property(PROPERTY warning_base
+  --diag_error=Pe223     # function "xxx" declared implicitly
+  --diag_warning=Pe054   # too few arguments in invocation of macro
+  --diag_warning=Pe144   # a value of type "void *" cannot be used to initialize an entity of type [...] "void (*)(struct onoff_manager *, int)"
+  --diag_warning=Pe167   # argument of type "void *" is incompatible with [...] "void (*)(void *, void *, void *)"
+  --diag_suppress=Pe1675 # unrecognized GCC pragma
+  --diag_suppress=Pe111  # statement is unreachable
+  --diag_suppress=Pe1143 # arithmetic on pointer to void or function type
+  --diag_suppress=Pe068  # integer conversion resulted in a change of sign)
+  )
+
+
+set(IAR_WARNING_DW_1   # >100
+  --diag_suppress=Pe188  # enumerated type mixed with another type
+  --diag_suppress=Pe128  # loop is not reachable
+  --diag_suppress=Pe550  # variable "res" was set but never used
+  --diag_suppress=Pe546  # transfer of control bypasses initialization
+  --diag_suppress=Pe186  # pointless comparison of unsigned integer with zero
+)
+set(IAR_WARNING_DW2
+  --diag_suppress=Pe1097 # unknown attribute
+  --diag_suppress=Pe381  # extra ";" ignored
+  --diag_suppress=Pa082  # undefined behavior: the order of volatile accesses is undefined
+  --diag_suppress=Pa084  # pointless integer comparison, the result is always false
+  --diag_suppress=Pe185  # dynamic initialization in unreachable code )
+  --diag_suppress=Pe167  # argument of type "onoff_notify_fn" is incompatible with...
+  --diag_suppress=Pe144  # a value of type "void *" cannot be used to initialize...
+  --diag_suppress=Pe177  # function "xxx" was declared but never referenced
+  --diag_suppress=Pe513  # a value of type "void *" cannot be assigned to an entity of type "int (*)(int)"
+)
+
+set(IAR_WARNING_DW3 )
+
+set_compiler_property(PROPERTY warning_dw_1
+  ${IAR_WARNING_DW_3}
+  ${IAR_WARNING_DW_2}
+  ${IAR_WARNING_DW_1} )
```

Comment:
no space at end and stick to one style, line 42 has ) on newline without indent, line 23 has it on newline with indent, update all to be the same